src/status_im/contexts/keycard/unblock/events.cljs

```diff
@@ -0,0 +1,89 @@
+             :pin        pin
+             :on-success (fn []
+                           (rf/dispatch [:keycard/disconnect])
+                           (rf/dispatch [:navigate-back])
+                           (rf/dispatch [:open-modal :screen/keycard.unblock.success]))}]]})))
+
+;; third stage, import keys
+(rf/reg-event-fx :keycard/unblock.import-keys
+ (fn []
+   {:fx [[:dispatch
+          [:keycard/connect.next-stage
+           {:on-error (fn [error]
+                        (if (= error :keycard/error.keycard-empty)
+                          (rf/dispatch [:keycard/unblock.generate-and-load-key])
+                          (do
+                            (rf/dispatch [:navigate-back])
+                            (rf/dispatch [:keycard/on-application-info-error error]))))}]]]}))
+
+;; second stage, check if card initialized
+(rf/reg-event-fx :keycard/unblock.init-card-or-import-keys
+ (fn [{:keys [db]}]
+   (let [pin                    (get-in db [:keycard :unblock :pin])
+         {:keys [initialized?]} (get-in db [:keycard :application-info])]
+     (if initialized?
+       (rf/dispatch [:keycard/unblock.import-keys])
+       (rf/dispatch
+        [:keycard/init-card
+         {:pin        pin
+          :on-success #(rf/dispatch [:keycard/unblock.import-keys])}])))))
+
+;; second stage, init card with pin
+(rf/reg-event-fx :keycard/unblock.init-with-pin
+ (fn []
+   {:fx [[:dispatch
+          [:keycard/connect.next-stage
+           {:on-error (fn [error]
+                        (if (= error :keycard/error.keycard-empty)
+                          (rf/dispatch [:keycard/unblock.init-card-or-import-keys])
+                          (do
+                            (rf/dispatch [:navigate-back])
+                            (rf/dispatch [:keycard/on-application-info-error error]))))}]]]}))
+
+;; first stage, reset card
+(rf/reg-event-fx :keycard/unblock
+ (fn [{:keys [db]}]
+   {:fx [[:dispatch
+          [:keycard/connect
+           {:key-uid  (get-in db [:keycard :application-info :key-uid])
+            :on-error (fn [error]
+                        (cond
+                          (or (= error :keycard/error.keycard-frozen)
+                              (= error :keycard/error.keycard-locked))
+                          (rf/dispatch
+                           [:keycard/factory-reset
+                            {:on-success #(rf/dispatch [:keycard/unblock.init-with-pin])}])
+
+                          (= error :keycard/error.keycard-empty)
+                          (rf/dispatch [:keycard/unblock.init-with-pin])
+
+                          :else
+                          (do
+                            (rf/dispatch [:navigate-back])
+                            (if (= error :keycard/error.keycard-wrong-profile)
+                              (do
+                                (rf/dispatch [:keycard/disconnect])
+                                (rf/dispatch [:open-modal :screen/keycard.different-card]))
```

Comment:
Could you please elaborate on why we need this here? Shouldn't this be part of on-application-info-error since we handle it in different flows?

(refs, https://github.com/status-im/status-mobile/pull/22008#issuecomment-2633417090, https://github.com/status-im/status-mobile/pull/22008#issuecomment-2633491898, https://github.com/status-im/status-mobile/issues/21647)

Reply:
good question, it's really confusing, probably the answer in your comment about logic. But it should be here, we don't have this screen in on-application-info-error

Reply:
> we don't have this screen in on-application-info-error

Shouldn't we add this screen there then, it will also fix these issues https://github.com/status-im/status-mobile/issues/22025 https://github.com/status-im/status-mobile/issues/21647